### PR TITLE
Refactoring of `FreezeTransitionLogic`

### PR DESCRIPTION
**Pull request by joan41868**
### Description:

This **PR** provides refactoring of the `FreezeTransitionLogic`, which freezes the network for a given timespan.

The refactoring's aim is to use `InvalidTransactionException` in order to abort invalid transactions.

### CODE REVIEW

File: hedera-node/src/main/java/com/hedera/services/legacy/handler/FreezeHandler.java [hunk 1]
```diff
@@ -61,24 +62,22 @@
 import static java.util.Calendar.MINUTE;
 
 /**
- * 		FreezeHandler is used in the HGCApp handleTransaction for performing Freeze
- * 		transactions. Documentation available at index.html#proto.FreezeTransactionBody
+ * FreezeHandler is used in the HGCApp handleTransaction for performing Freeze
+ * transactions. Documentation available at index.html#proto.FreezeTransactionBody
```
Comment (georgiyazovaliiski) [inline hedera-node/src/main/java/com/hedera/services/legacy/handler/FreezeHandler.java:66]: Nit: Can we undo this.

File: hedera-node/src/test/java/com/hedera/services/txns/network/FreezeTransitionLogicTest.java [hunk 2]
```diff
@@ -91,7 +91,7 @@ void capturesBadFreeze() {
 						.build())
 				.build();
 		givenTxnCtx();
-		given(delegate.perform(freezeTxn, consensusTime)).willReturn(freezeRec);
+//		given(delegate.perform(freezeTxn, consensusTime)).willReturn(freezeRec);
```
Comment (georgiyazovaliiski) [inline hedera-node/src/test/java/com/hedera/services/txns/network/FreezeTransitionLogicTest.java:94]: Why is this commented? Should it be replaced by some other assertion?

File: hedera-node/src/test/java/com/hedera/services/txns/network/FreezeTransitionLogicTest.java [hunk 3]
```diff
@@ -106,7 +106,7 @@ void followsHappyPathWithOverrides() {
...
```
Comment (georgiyazovaliiski) [inline hedera-node/src/test/java/com/hedera/services/txns/network/FreezeTransitionLogicTest.java:109]: Why is this commented? Should it be replaced by some other assertion?

File: hedera-node/src/test/java/com/hedera/services/txns/network/FreezeTransitionLogicTest.java [hunk 4]
```diff
@@ -159,7 +159,7 @@ void rejectsMissingFileHash() {
 	@Test
 	void translatesUnknownException() {
 		givenTxnCtx();
-		given(delegate.perform(any(), any())).willThrow(IllegalStateException.class);
+//		given(delegate.perform(any(), any())).willThrow(IllegalStateException.class);
```
Comment (georgiyazovaliiski) [inline hedera-node/src/test/java/com/hedera/services/txns/network/FreezeTransitionLogicTest.java:162]: Why is this commented? Should it be replaced by some other assertion?

**Review by georgiyazovaliiski — CHANGES_REQUESTED** [review]

**Review by CordonaCodeCraft — APPROVED** [review]
OLGTM

**Pull request merged**